doc/nrf/security.rst

```diff
@@ -1,17 +1,56 @@
+.. _security:
 .. _security_index:
 
 Security
 ########
 
-The |NCS| provides a variety of security features.
-Make sure to check the :ref:`overview of available options <security>` and consider enabling :ref:`access port protection mechanism <app_approtect>`.
+This section provides an overview of core security features available in Nordic Semiconductor products.
+The features are made available either as built-ins in modules, drivers, and subsystems, or are shown in samples or applications in |NCS|.
 
-If you are working with nRF53 Series or nRF91 Series devices, use :ref:`Trusted Firmware-M <ug_tfm>` to increase the security of your end product.
+The following table lists the available general security features.
+Some of them are documented in detail in other parts of this documentation, while others are documented in the subpages in this section.
+
+ .. list-table::
+  :header-rows: 1
+
+  * - Security feature
+    - Description
+    - Configuration
+    - Available dedicated components
+  * - Access port protection (AP-Protect)
+    - When enabled, this mechanism blocks the debugger from read and write access to all CPU registers and memory-mapped addresses.
+    - See :ref:`app_approtect`.
+    - ---
+  * - Bootloader and Device Firmware Upgrade (DFU)
+    - The |NCS| supports :ref:`MCUboot and nRF Secure Immutable Bootloader (NSIB) <ug_bootloader_mcuboot_nsib>` for secure boot, and DFU procedures using MCUboot and :ref:`Software Updates for Internet of Things (SUIT) <ug_nrf54h20_suit_dfu>`.
+    - See :ref:`app_bootloaders`.
+    - :ref:`nRF Secure Immutable Bootloader (NSIB) <bootloader>`, :ref:`DFU libraries <lib_dfu>`, :ref:`Software Updates for Internet of Things (SUIT) <ug_nrf54h20_suit_dfu>`, :doc:`MCUboot <mcuboot:design>`
+  * - Trusted Firmware-M (TF-M)
+    - TF-M is the reference implementation of `Platform Security Architecture (PSA)`_.
+      On nRF5340 and nRF91 Series devices, TF-M is used to configure and boot an application as non-secure.
+    - See :ref:`ug_tfm`.
+    - :ref:`tfm_samples`, :ref:`cryptography samples <crypto_samples>`, :ref:`https_client` sample, :ref:`openthread_samples`, :ref:`TF-M integration samples <zephyr:tfm_integration-samples>` in Zephyr
```

Comment:
I am suspecting that we have more samples using TF-M these days. Both cellular, Bluetooth, Mesh, and possibly Wi-Fi uses it in samples with *ns board targets.
This is more of a note to self, I need to update both here and in ug_tfm most likely :)

Reply:
But doesn't the `*/ns` target refer to the processing environments and security by separation implemented primarily through ARM TrustZone? https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/app_dev/board_support/processing_environments.html This separation is then using TF-M, yes, but in the past we had SPM, so I see `*/ns` as belonging to a different realm.